### PR TITLE
Correct Boms

### PR DESCRIPTION
**Pull request by robert3005**
@jared2501 this should be what you wanted in #55 
@ash211 previous pr was unfortunately off since it was missing dependencyManagement section. 

also @dansanduleac 

Tested via local publish and resolution in another project

### CODE REVIEW

**Comment by ash211** [comment]
Tests flaked (https://github.com/palantir/spark/issues/134) so rerunning

**Comment by robert3005** [comment]
`include an external JAR in SparkR - org.apache.spark.deploy.SparkSubmitSuite` might be a legit failure, checking.

**Comment by robert3005** [comment]
Alright, this is a flake

File: dists/hadoop-palantir/pom.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <parent>
+    <groupId>org.apache.spark</groupId>
+    <artifactId>spark-parent_2.11</artifactId>
+    <version>2.2.0-SNAPSHOT</version>
```
Comment (dansanduleac) [inline dists/hadoop-palantir/pom.xml]: Can this use `${project.version}` ? And scala.binary.version in the artifactid

**Comment by robert3005** [comment]
Can't repro this locally. Will have to dig more

File: dists/hadoop-palantir/pom.xml (hunk 1 above)
Comment (robert3005) [inline dists/hadoop-palantir/pom.xml]: This is not possible. Anything here has to be resolvable at configuration time and project.version isn't configured at that point. These get replaced by publish steps.

**Pull request merged**